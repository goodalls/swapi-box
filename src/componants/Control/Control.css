.control {
  grid-column: 1/13;
  grid-row: 2;
  display: flex;
  justify-content: center;
  height: 50px; }

button {
  background: yellow;
  border-radius: 28px;
  color: #000;
  font-size: 20px;
  padding: 10px 20px 10px 20px;
  text-decoration: none;
  margin: 0 1%; }

button:hover {
  background: #3cb0fd;
  text-decoration: none; }
