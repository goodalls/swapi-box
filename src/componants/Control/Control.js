import React from 'react';
import './Control.css';
import PropTypes from 'prop-types';

const Control = ({ favorites, cards, active }) => {
  return (
    <div className="control">
      <button
        onClick={cards}
        name="people"
        className={active === 'people' ? 'activeClass' : 'notActive'}
      >
        People
      </button>
      <button
        onClick={cards}
        name="planets"
        className={active === 'planets' ? 'activeClass' : 'notActive'}
      >
        Planets
      </button>
      <button
        onClick={cards}
        name="vehicles"
        className={active === 'vehicles' ? 'activeClass' : 'notActive'}
      >
        Vehicles
      </button>
      <button
        name="favorites"
        className={active === 'favorites' ? 'activeClass' : 'notActive'}
      >
        Favorites <span className="favorite">{favorites}</span>
      </button>
    </div>
  );
};

Control.propTypes = {
  favorites: PropTypes.number,
  people: PropTypes.func,
  active: PropTypes.string
};
export default Control;
