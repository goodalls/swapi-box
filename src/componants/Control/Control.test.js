import React from 'react';
import Control from './Control';
import { shallow } from 'enzyme';

describe('Control', () => {
  it('should match snapshot', () => {
    const wrapper = shallow(<Control />);
    expect(wrapper).toMatchSnapshot();
  });
});
