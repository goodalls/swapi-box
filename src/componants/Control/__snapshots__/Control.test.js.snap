// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Control should match snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Control />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <button
          className="notActive"
          name="people"
          onClick={undefined}
>
          People
</button>,
        <button
          className="notActive"
          name="planets"
          onClick={undefined}
>
          Planets
</button>,
        <button
          className="notActive"
          name="vehicles"
          onClick={undefined}
>
          Vehicles
</button>,
        <button
          className="notActive"
          name="favorites"
          onClick={undefined}
>
          Favorites 
          <span
                    className="favorite"
          />
</button>,
      ],
      "className": "control",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "People",
          "className": "notActive",
          "name": "people",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "People",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Planets",
          "className": "notActive",
          "name": "planets",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Planets",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vehicles",
          "className": "notActive",
          "name": "vehicles",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": "Vehicles",
        "type": "button",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            "Favorites ",
            <span
              className="favorite"
/>,
          ],
          "className": "notActive",
          "name": "favorites",
          "onClick": undefined,
        },
        "ref": null,
        "rendered": Array [
          "Favorites ",
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "favorite",
            },
            "ref": null,
            "rendered": null,
            "type": "span",
          },
        ],
        "type": "button",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <button
            className="notActive"
            name="people"
            onClick={undefined}
>
            People
</button>,
          <button
            className="notActive"
            name="planets"
            onClick={undefined}
>
            Planets
</button>,
          <button
            className="notActive"
            name="vehicles"
            onClick={undefined}
>
            Vehicles
</button>,
          <button
            className="notActive"
            name="favorites"
            onClick={undefined}
>
            Favorites 
            <span
                        className="favorite"
            />
</button>,
        ],
        "className": "control",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "People",
            "className": "notActive",
            "name": "people",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "People",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Planets",
            "className": "notActive",
            "name": "planets",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Planets",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vehicles",
            "className": "notActive",
            "name": "vehicles",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": "Vehicles",
          "type": "button",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Favorites ",
              <span
                className="favorite"
/>,
            ],
            "className": "notActive",
            "name": "favorites",
            "onClick": undefined,
          },
          "ref": null,
          "rendered": Array [
            "Favorites ",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": undefined,
                "className": "favorite",
              },
              "ref": null,
              "rendered": null,
              "type": "span",
            },
          ],
          "type": "button",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
